From c9e763461847db10d4b828ecfc18cbd123df7595 Mon Sep 17 00:00:00 2001
From: maxina <maxina@motorola.com>
Date: Fri, 30 Oct 2015 13:55:16 +0800
Subject: [PATCH 438/959] IKSWM-7127: usb: phy: Add a module param to disable
 host mode

Add a module param to disable host mode for factory to test the
pin routing.
echo 1 > /sys/module/phy_msm_usb/parameters/host_mode_disable
disable host mode

Change-Id: I0e6dd46d700f9f75b55466edfca490898949c762
Signed-off-by: maxina <maxina@motorola.com>
Reviewed-on: http://gerrit.mot.com/795994
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Ravikumar Vembu <raviv@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/usb/phy/phy-msm-usb.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/phy/phy-msm-usb.c b/drivers/usb/phy/phy-msm-usb.c
index 0609f55f38e..6afb83c2ef7 100644
--- a/drivers/usb/phy/phy-msm-usb.c
+++ b/drivers/usb/phy/phy-msm-usb.c
@@ -103,6 +103,11 @@ module_param(floated_charger_enable , bool, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(floated_charger_enable,
 	"Whether to enable floated charger");
 
+static bool host_mode_disable;
+module_param(host_mode_disable , bool, S_IRUGO | S_IWUSR);
+MODULE_PARM_DESC(host_mode_disable,
+	"Whether to disable Host Mode");
+
 /* by default debugging is enabled */
 static unsigned int enable_dbg_log = 1;
 module_param(enable_dbg_log, uint, S_IRUGO | S_IWUSR);
@@ -3310,7 +3315,9 @@ static void msm_otg_init_sm(struct msm_otg *motg)
 			else
 				clear_bit(B_SESS_VLD, &motg->inputs);
 		} else if (pdata->otg_control == OTG_PMIC_CONTROL) {
-			if (pdata->pmic_id_irq) {
+			if (host_mode_disable)
+				set_bit(ID, &motg->inputs);
+			else if (pdata->pmic_id_irq) {
 				if (msm_otg_read_pmic_id_state(motg))
 					set_bit(ID, &motg->inputs);
 				else
@@ -4460,6 +4467,11 @@ static void msm_id_status_w(struct work_struct *w)
 
 	dev_dbg(motg->phy.dev, "ID status_w\n");
 
+	if (host_mode_disable) {
+		dev_dbg(motg->phy.dev, "Ignore ID - host mode is disabled\n");
+		return;
+	}
+
 	if (motg->pdata->pmic_id_irq)
 		motg->id_state = msm_otg_read_pmic_id_state(motg);
 	else if (motg->ext_id_irq)
-- 
2.11.0

